GameDataParser/Program.cs

```diff
@@ -3,87 +3,51 @@
-                Thread prestigeThread = new Thread(() => PrestigeMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread expThread = new Thread(() => ExpMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread questThread = new Thread(() => QuestMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread scriptThread = new Thread(() => ScriptMetadataExport.Export(xmlFiles, xmlMemFile));
-                Thread guildThread = new Thread(() => GuildMetadataExport.Export(xmlFiles, xmlMemFile));
+            CultureInfo.CurrentCulture = new CultureInfo("en-US");
+            MetadataResources resources = new MetadataResources();
 
-                Spinner spinner = new Spinner();
-                spinner.Start();
+            IEnumerable<MetadataExporter> exports = new List<MetadataExporter>()
+            {
+                new ItemParser(resources),
+                new MapEntityParser(resources),
+                new SkillParser(resources),
+                new InsigniaParser(resources),
+                new ExpTableParser(resources),
+                new QuestParser(resources),
+                new ScriptParser(resources),
+                new GuildParser(resources),
+                new PrestigeParser(resources)
+            };
 
-                itemThread.Start();
-                mapEntityThread.Start();
-                skillThread.Start();
-                insigniaThread.Start();
-                prestigeThread.Start();
-                expThread.Start();
-                questThread.Start();
-                scriptThread.Start();
-                guildThread.Start();
+            IEnumerable<Task> tasks = exports.Select(exporter => Task.Run(() => exporter.export()));
 
-                itemThread.Join();
-                mapEntityThread.Join();
-                skillThread.Join();
-                insigniaThread.Join();
-                prestigeThread.Join();
-                expThread.Join();
-                questThread.Join();
-                scriptThread.Join();
-                guildThread.Join();
+            await Task.WhenAll(tasks);
 
-                spinner.Stop();
+            spinner.Stop();
+            TimeSpan runtime = spinner.getRuntime();
 
-                Console.WriteLine("\r\nFinished parsing.");
-            }
-            catch (Exception ex)
-            {
-                if (ex.ToString().Contains("Xml.m2h") || ex.ToString().Contains("Xml.m2d") || ex.ToString().Contains("Exported.m2h") || ex.ToString().Contains("Exported.m2d"))
-                {
-                    Console.ForegroundColor = ConsoleColor.Red;
-                    Console.WriteLine("Error: Missing pack files.\nPlease copy all pack files (Xml.m2d, Xml.m2h, Exported.m2d, Exported.m2h) from game client into GameDataParser/Resources.");
-                }
-            }
+            Console.WriteLine("\rExporting finished in " + runtime.Minutes + " minutes and " + runtime.Seconds + " seconds");
```

Comment:
Can we just use a decimal value here? Ex: 9 minutes and 30 seconds => 9.5 minutes.

Reply:
Why? I'd have to do some math round/floor to support this nicely, and for basically no real upside

Reply:
That's true, I suppose the current implementation is fine or can simply convert to use `StopWatch.ElapsedMilliseconds` instead of `StopWatch.Elapsed`